food-vendor/src/main/java/com/example/foodvendor/data/FoodVendorDAO.java

```diff
@@ -20,31 +22,30 @@
 
     private Map<Integer, Map<String, Ingredient>> vendorToIngredientsMap;
 
-    private static final Tracer tracer = Tracing.getTracer();
+    private static final SpanProcessor spanProcessor =
+            SimpleSpansProcessor.create(new LoggingSpanExporter());
+    private static final Tracer tracer = OpenTelemetrySdk.getTracerProvider().get("");
 
     public FoodVendorDAO() {
         setupVendorToIngredientsMap();
     }
 
     public Ingredient getIngredient(int id, String ingredient) {
-        Span span = tracer.spanBuilder("FoodVendor starting query...")
-                .setRecordEvents(true)
-                .setSampler(Samplers.alwaysSample())
-                .startSpan();
-
-        Map<String, Ingredient> vendorInventory = vendorToIngredientsMap.get(id);
-        if (vendorInventory == null) {
+        OpenTelemetrySdk.getTracerProvider().addSpanProcessor(spanProcessor);
+        Span span = tracer.spanBuilder("FoodVendor starting query...").startSpan();
+
+        try (Scope scope = tracer.withSpan(span)) {
+            Map<String, Ingredient> vendorInventory = vendorToIngredientsMap.get(id);
+            if (vendorInventory == null) {
+                span.end();
```

Comment:
We will call span.end() twice (once here and one in the finally clause). Is that intentional?

Reply:
Nice catch, removed the extra call